src/resolvers/contractPerils.ts

```diff
@@ -1,21 +1,29 @@
 import { Upstream } from '../api/upstream'
 import {
-  Locale,
+  ContractContractPerilsArgs,
+  ContractResolvers,
   PerilV2,
+  QueryContractPerilsArgs,
   QueryResolvers,
-  TypeOfContract,
 } from '../generated/graphql'
 
-interface PerilsArgs {
-  locale: Locale
-  contractType: TypeOfContract
-}
-
 export const contractPerils: QueryResolvers['contractPerils'] = async (
   _,
-  args: PerilsArgs,
+  args: QueryContractPerilsArgs,
   { upstream }: { upstream: Upstream },
 ): Promise<PerilV2[]> => {
   const { perils } = await upstream.promiseCms.getContractInfo(args)
   return perils
 }
+
+export const contractPerilExtension: ContractResolvers['contractPerils'] = async (
+  parent,
+  args: ContractContractPerilsArgs,
+  { upstream }: { upstream: Upstream },
```

Comment:
```suggestion
  { upstream },
```

And then not import `Upstream`?